### ISSUE TITLE
Image drag and drop task in Lumi does not create correct answers

### ISSUE THREAD

**Issue by natobsid** [Lumieducation/Lumi#1945]
**Describe the bug**

Text elements added to drop zones are not recognised as correct. An analysis of the content.json in comparison to the same task generated through the h5p moodle plugin shows that elements marked in the drop zone are not being correctly marked as `"correctElements":["0"]` but as `"correctElements":[]`, causing them to be interpreted as incorrect answers . Manually adding ["0"] to a drop zone in content.json solves the problem. The issue is generated regardless of whether the image drag and drop task is created as a standalone or within a nested element such as a course presentation.

Drop zone code generated by LUMI:
`{"x":16.948711700726808,"y":30.897249033711034,"width":1.5,"height":1.5,"correctElements":[],"showLabel":false,"backgroundOpacity":0,"tipsAndFeedback":{"tip":""},"single":false,"autoAlign":true,"type":{"library":"H5P.DragQuestionDropzone 0.1"},"label":"<div>1</div>\n"},`

Drop zone code with 'correct item' marked:

`{"x":16.948711700726808,"y":30.897249033711034,"width":1.5,"height":1.5,"correctElements":["0"],"showLabel":false,"backgroundOpacity":0,"tipsAndFeedback":{"tip":""},"single":false,"autoAlign":true,"type":{"library":"H5P.DragQuestionDropzone 0.1"},"label":"<div>1</div>\n"},`

 

**Environment (please complete the following information):**
 - OS: MacOS BigSur 11.2
 - Version 0.8.2 (0.8.2)

**Expected behavior**
Text elements added to drop zones should be scored as 'correct'.

**Comment by sr258** [comment]
Thank you for your very detailled bug report! I am able to reproduce this and it's already a problem in our core library (h5p-nodejs-library), but the incorrect data structure already occurs when the H5P editor sends the content JSON structure to our server: 
![image](https://user-images.githubusercontent.com/15268740/139592151-440fedd2-76d4-4544-9e3f-750f40693b7b.png)

Either something is wrong in the library structure the editor receives from our server implemention so that it doesn't set the correct elements. Or it might be a bug in the content type's editor code... Can you confirm the the version of the content type you use on Moodle is exactly the same as the most recent one (H5P.DragQuestion 1.13.14 and H5PEditor.DragQuestion 1.10.16)?

**Comment by natobsid** [comment]
Hi Sebastian,

Happy to help — I think you’re doing really great work and something which could make H5P a lot more accessible to teachers in future.  

I can’t confirm the content type version, I’m afraid, as I am not admin on our site, but I can confirm that the h5p hub plugin is up to date (1.22.3). In any case, wouldn’t the fact that this is taking place at a local level in Lumi suggest that it is not a library content issue on the server side. When I preview in Lumi I get the same issue: correct answers are marked wrong.

Thanks for your time and please let me know if I can do anything else to help regarding this issue.

Nathan

**Comment by sr258** [comment]
Lumi includes a server under the hood, even though it’s not obvious to the end user. When you save content in Lumi it creates a request to the built-in server.

Can you post a h5p file created by Moodle (you might have to rename it to .zip for this to work) and I‘ll check versions there.

**Comment by sr258** [comment]
@natobsid I've tried this with a Drupal 7 installation of H5P and I get the same error there. So this is a bug in the content type, not in Lumi. I'll report it to H5P's dev. The report is here: https://h5p.org/node/1224905

**Comment by natobsid** [comment]
Thanks @sr258  — that’s good to know. Funny how it works on my organisation's moodle, though. Sorry I haven’t been able to get back earlier on your request. I can still send you a couple of comparison h5p files if you feel that would help, but as I don’t have access to the libraries on our moodle, I don’t know if it would, as you’d see the final code but not at server side. Let me know, though.

Anyway, I CAN confirm that server-side on Lumi, I do have  H5P.DragQuestion 1.13.14 and H5PEditor.DragQuestion 1.10.16, so it seems to be up to date.

Best,

Nathan

**Comment by sr258** [comment]
The h5p files you can download from moodle contain all the required libraries, so some samples would help.

**Comment by gilhoo** [comment]
Hi,
I have exactly the same issue. Using Lumi, Logiquiz or h5p implemented on my school Moodle.
I could fix the problem by editing my h5p file in h5p.org

**Comment by sr258** [comment]
If you delete the content type and its editor in the Lumi library management panel, you can then upload a file you've downloaded from a source where things still work. The old library will then be usable to create new content in Lumi. The bug should not occur then.

**Comment by natobsid** [comment]
Thanks! That’s done the trick—I can confirm that this workaround works for me. Had a bit of trouble deleting the libraries though due to co-dependencies. Just in case anyone is having trouble with this, what I ended up doing was deleting the app and reinstalling. I then took a working drag and drop that I had created in my moodle with older library files and opened it in lumi, which meant that these libraries were imported, as Sebastian has stated. 

Be careful to do this right away if you are planning on using drag and drop in any other task type (such as interactive video/course presentation)—if you add a drag and drop and haven’t imported your old libraries you will automatically get the new ones!

Thanks Sebastian for this temporary fix—let’s hope h5p gets onto it soon.

Best Nathan
